### PR TITLE
Fix exec

### LINKED ISSUE #4108: Dashboard 2.0.0-beta2 has non-working exec
##### Environment

```
Installation method: changed version of dashboard docker image to beta2 in deployment for beta1
Kubernetes version: 1.15.0
Dashboard version: 2.0.0-beta2
Operating system: Debian 9
Node.js version ('node --version' output):
Go version ('go version' output):
```

##### Steps to reproduce

1. Choose deployment
2. Choose pod of deployment
3. Click 'Exec' into pod
##### Observed result

Black screen of terminal that hasn't any shell prompt

### PR DESCRIPTION
**Pull request by floreks**
Fixes #4108 

I have slightly refactored exec and added utility service to handle things like exec, logs, etc.

### CODE REVIEW

**Comment by floreks** [comment]
@slavditore @pog2g @lg-sing @mausch @alx-kmn once this gets merged and [head image](https://github.com/kubernetes/dashboard/blob/master/aio/deploy/head.yaml) gets pushed can you test if it works for you?

File: src/app/frontend/shell/component.ts [hunk 1]
```diff
@@ -175,20 +193,26 @@ export class ShellComponent implements AfterViewInit, OnDestroy {
   }
 
   private async setupConnection(): Promise<void> {
+    console.log('before setup conn');
     if (!(this.selectedContainer && this.podName && this.namespace_ && !this.connecting_)) {
       return;
     }
 
+    console.log('connecting');
```
Comment (maciaszczykm) [inline src/app/frontend/shell/component.ts]: Please remove the console logs.

File: src/app/frontend/shell/component.ts [hunk 2]
```diff
@@ -175,20 +193,26 @@ export class ShellComponent implements AfterViewInit, OnDestroy {
   }
 
   private async setupConnection(): Promise<void> {
+    console.log('before setup conn');
```
Comment (maciaszczykm) [inline src/app/frontend/shell/component.ts]: As before.

**Comment by ld-singh** [comment]
@floreks the head image works fine for me...!!!! Thanks

**Pull request merged**